lib/monkey/mk_server/mk_http_parser.c

```diff
@@ -78,6 +78,21 @@ struct row_entry mk_headers_table[] = {
     { 10, "user-agent"          }
 };
 
+static inline void reverse_char_lookup(char *buf, char c, int len, struct mk_http_parser *p)
+{
+    int x = 0;
+    int y = 0;
+
+    x = p->i;
+    do {
+        if (buf[x - y] == c) {
+            p->i = x - y;
+            return;
+        }
+        y++;
+    } while (y < len);
```

Comment:
Just curious,
(a) isn't this effectively reimplementing [`memrchr()`](https://linux.die.net/man/3/memrchr)? This could be:
```c
    char *r = (char*)memrchr(buf + p->i - len, c, (size_t)len);
    if (r) {
        p->i = r - buf;
        return;
    }
```
(b) why isn't this a `for` loop, i.e.:
```c
    x = p->i;
    for (int y = 0; y < len; ++y) {
        if (buf[x - y] == c) {
            p->i = x - y;
            return;
        }
    }
```
or, better yet:
```c
    int lim = p->i - len;
    for (int x = p->i; x > lim; --x) {
        if (buf[x] == c) {
            p->i = x;
            return;
        }
    }
```
?

Reply:
cc: @leonardo-albertovich

Reply:
note: those changes comes from https://github.com/monkey/monkey

Reply:
I made it that way to mimic the existing code to make it easier to understand (considering how char_lookup is implemented).

Note: I noticed that memchr is available according to msdn but couldn't find anything about memrchr so if we consider changing those 2 functions we might want to double check that (otherwise we'll end up with the same code just used in a smaller portion of the installs which could lead to hard to trace bugs) (this is just food for thought)